### ISSUE TITLE
bug(8.2-alpha): Android frame transition throws exception

### ISSUE THREAD

**Issue by DimitrisRK** [NativeScript/NativeScript#9732]
### Issue Description

In android, when I navigate from a page to another programmatically, I receive an exception (output provided).

Exception seems to occur because of this line here: https://github.com/NativeScript/NativeScript/blob/release/8.2.0/packages/core/ui/frame/index.android.ts#L1028

### Reproduction

Navigate among pages programmatically. This occured to me on 3rd navigation.
One of my navigations run inside `onNavigatingTo` event callback of current page.

### Relevant log output (if applicable)

```shell
System.err: An uncaught Exception occurred on "main" thread.
System.err: Calling js method onDestroy failed
System.err: Error: fragment0[0]<null>.onDestroy: entry has no resolvedPage
System.err: 
System.err: StackTrace:
System.err: error(file: node_modules/@nativescript/core/trace/index.js:248:0)
System.err:     at onDestroy(file: node_modules/@nativescript/core/ui/frame/index.android.js:818:12)
System.err:     at onDestroy(file: node_modules/@nativescript/core/ui/frame/fragment.android.js:38:0)
System.err:     at com.tns.Runtime.callJSMethodNative(Native Method)
System.err:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1302)
System.err:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:1188)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1175)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1153)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1149)
System.err:     at com.tns.FragmentClass.onDestroy(FragmentClass.java:86)
System.err:     at androidx.fragment.app.Fragment.performDestroy(Fragment.java:2928)
System.err:     at androidx.fragment.app.FragmentStateManager.destroy(FragmentStateManager.java:492)
System.err:     at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1308)
System.err:     at androidx.fragment.app.FragmentManager.removeCancellationSignal(FragmentManager.java:743)
System.err:     at androidx.fragment.app.FragmentManager$2.onComplete(FragmentManager.java:370)
System.err:     at androidx.fragment.app.FragmentTransition$3.run(FragmentTransition.java:408)
System.err:     at androidx.transition.FragmentTransitionSupport$5.onTransitionEnd(FragmentTransitionSupport.java:280)
System.err:     at androidx.transition.Transition.end(Transition.java:1965)
System.err:     at androidx.transition.TransitionSet$TransitionSetListener.onTransitionEnd(TransitionSet.java:451)
System.err:     at androidx.transition.Transition.end(Transition.java:1965)
System.err:     at androidx.transition.Transition$3.onAnimationEnd(Transition.java:1914)
System.err:     at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:554)
System.err:     at android.animation.ValueAnimator.endAnimation(ValueAnimator.java:1250)
System.err:     at android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1492)
System.err:     at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
System.err:     at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
System.err:     at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
System.err:     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:973)
System.err:     at android.view.Choreographer.doCallbacks(Choreographer.java:799)
System.err:     at android.view.Choreographer.doFrame(Choreographer.java:730)
System.err:     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:960)
System.err:     at android.os.Handler.handleCallback(Handler.java:938)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:99)
System.err:     at android.os.Looper.loop(Looper.java:236)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:8057)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:620)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1011)
Successfully synced application com.fuelics.mobile.alpha on device 7pizbqwkhexkw85h.
```

### Environment

@nativescript/core: 8.2.0-alpha.1

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by DimitrisRK** [comment]
First post updated with more accurate details.

**Comment by NathanWalker** [comment]
closed via 8.2.0-alpha.4
